File: backend/src/main/java/com/redscooter/API/order/DTO/GetOrderDTO.java
package com.redscooter.API.order.DTO;

import com.redscooter.API.OrderLine.DTO.GetOrderLineDTO;
import com.redscooter.API.OrderLine.OrderLine;
import com.redscooter.API.common.AuditBaseDTO;
import com.redscooter.API.order.Order;
import com.redscooter.API.order.OrderStatus;
import lombok.Getter;
import lombok.Setter;

import java.util.List;
import java.util.stream.Collectors;


@Getter
@Setter
public class GetOrderDTO extends AuditBaseDTO {
    private Long id;
    private OrderStatus orderStatus;
    private List<GetOrderLineDTO> orderLines;

    private GetOrderBillingDTO orderBilling;

    public GetOrderDTO(Order o) {
        id = o.getId();
        orderStatus = o.getOrderStatus();
        orderLines = o.getOrderLines().stream().map(ol -> OrderLine.toGetOrderLineDTO(ol)).collect(Collectors.toList());
        orderBilling = o.getOrderBilling().toGetOrderBillingDTO();

        this.createdAt = o.getCreatedAt();
        this.updatedAt = o.getUpdatedAt();
    }
}

File: frontend/src/services/models/GetOrderDTO.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { GetOrderBillingDTO } from './GetOrderBillingDTO';
import type { GetOrderLineDTO } from './GetOrderLineDTO';

export type GetOrderDTO = {
    createdAt?: number;
    updatedAt?: number;
    id?: number;
    orderStatus?: GetOrderDTO.orderStatus;
    orderLines?: Array<GetOrderLineDTO>;
    orderBilling?: GetOrderBillingDTO;
};

export namespace GetOrderDTO {

    export enum orderStatus {
        NEW = 'NEW',
        KONFIRMUAR = 'KONFIRMUAR',
        N_D_RGIM = 'NË DËRGIM',
        REFUZUAR = 'REFUZUAR',
        MBYLLUR = 'MBYLLUR',
    }


}
